feature-libs/product-configurator/rulebased/components/group-title/configurator-group-title.component.ts

```diff
@@ -41,31 +50,67 @@ export class ConfiguratorGroupTitleComponent {
   iconTypes = ICON_TYPE;
 
   //TODO(CXSPA-1014): make ConfiguratorExpertModeService a required dependency
+  /**
+   * @deprecated since 5.1
+   */
   constructor(
     configuratorCommonsService: ConfiguratorCommonsService,
     configuratorGroupsService: ConfiguratorGroupsService,
-    configRouterExtractorService: ConfiguratorRouterExtractorService,
-    // eslint-disable-next-line @typescript-eslint/unified-signatures
-    configExpertModeService: ConfiguratorExpertModeService
+    configRouterExtractorService: ConfiguratorRouterExtractorService
   );
 
-  /**
-   * @deprecated since 5.1
-   */
   constructor(
     configuratorCommonsService: ConfiguratorCommonsService,
     configuratorGroupsService: ConfiguratorGroupsService,
-    configRouterExtractorService: ConfiguratorRouterExtractorService
+    configRouterExtractorService: ConfiguratorRouterExtractorService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    configExpertModeService: ConfiguratorExpertModeService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    breakpointService: BreakpointService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    configuratorStorefrontUtilsService: ConfiguratorStorefrontUtilsService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    hamburgerMenuService: HamburgerMenuService
   );
 
   constructor(
     protected configuratorCommonsService: ConfiguratorCommonsService,
     protected configuratorGroupsService: ConfiguratorGroupsService,
     protected configRouterExtractorService: ConfiguratorRouterExtractorService,
     @Optional()
-    protected configExpertModeService?: ConfiguratorExpertModeService
+    protected configExpertModeService?: ConfiguratorExpertModeService,
+    @Optional()
+    protected breakpointService?: BreakpointService,
+    @Optional()
+    protected configuratorStorefrontUtilsService?: ConfiguratorStorefrontUtilsService,
+    @Optional()
+    protected hamburgerMenuService?: HamburgerMenuService
   ) {}
 
+  ngOnInit(): void {
+    this.subscription.add(
```

Comment:
It would be more clear to only subscribe if both services are present, do you agree?

Reply:
Agree! Done!